VirtoCommerce.Storefront/Infrastructure/Autorest/UserPasswordAuthHandler.cs

```diff
@@ -97,21 +95,6 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
             return await base.SendAsync(request, cancellationToken).ConfigureAwait(false);
         }
 
-        /// <summary>
-        /// Releases the unmanaged resources used by the <see cref="T:System.Net.Http.DelegatingHandler" />, and optionally disposes of the managed resources.
-        /// </summary>
-        /// <param name="disposing">true to release both managed and unmanaged resources; false to releases only unmanaged resources.</param>
-        protected override void Dispose(bool disposing)
-        {
-            if (disposing && !_disposed)
-            {
-                _disposed = true;
-                _lock.Dispose();
-            }
-
-            base.Dispose(disposing);
-        }
-
         private async Task<bool> RenewTokensAsync(CancellationToken cancellationToken)
         {
             if (await _lock.WaitAsync(Timeout, cancellationToken).ConfigureAwait(false))
```

Comment:
Here are 2 problems:
1. Platform v2 token endpoint seems to be "/token", not "connect/token". I think need to unify with 3.0 or make it configurable.
2. If we set Endpoint Url in options like "http://localhost/admin" (without trailing slash), we are getting incorrect url for token endpoint here. Probably need to handle such situations, e.g.:
```
                    var client = _clientFactory.CreateClient();
                    var endpointUrl = _options.Url.ToString();

                    if (!endpointUrl.EndsWith('/'))
                    {
                        endpointUrl += '/';
                    }

                    var response = await client.RequestPasswordTokenAsync(new PasswordTokenRequest { Address = endpointUrl + "token", UserName = _options.UserName, Password = _options.Password });

```

Reply:
1. It is Ok that it works only for platform v3.
2. Fixed url part combining.